### ISSUE TITLE
[SPIR-V] `[[vk::combinedImageSampler]]` refuses to work with `SamplerComparisonState`

### ISSUE THREAD

**Issue by h3r2tic** [microsoft/DirectXShaderCompiler#4696]
Hey, I tried doing this:

```hlsl
[[vk::combinedImageSampler]]
Texture2D foo: register(t0);
[[vk::combinedImageSampler]]
SamplerComparisonState foo_sampler: register(s0);
```

And got this error:

> 'combinedImageSampler' attribute only applies to Textures (e.g., Texture2D) and SamplerState

Interestingly, there is a workaround, which even causes SPIR-V Cross to produce the correct GLSL 😀: make a dummy `SamplerState` using the same register, and apply [[vk::combinedImageSampler]] to that instead!

```hlsl
[[vk::combinedImageSampler]]
Texture2D foo: register(t0);
[[vk::combinedImageSampler]]
SamplerState foo_sampler_UNUSED: register(s0);
SamplerComparisonState foo_sampler: register(s0);
```

**Comment by StarsX** [comment]
Any update for this issue? It seems even the workaround no longer works in the latest release.

**Comment by sudonatalie** [comment]
Thanks for reporting, and sorry that this issue went without response for so long! This should be fixed now with #6338 .
